abbr/abbr.py

```diff
@@ -59,9 +59,44 @@ def findall(text):
                         abbrevs[abb] = None
                 else:
                     abbrevs[abb] = None
+                    logger.info('No full term detected for '
+                                'abbreviation {0}'.format(abb))
+        else:
+            logger.warning('Abbreviation detection regex returned None.')
     return abbrevs
 
 
+def compressall(text):
+    """
+    Search for abbreviations in text using re_abbr (defined in utils.get_res).
+    For each abbreviation, find likely full term. Replace each instance of the
+    full term in the text with the abbreviation.
+
+    Parameters
+    ----------
+    text : str
+        Text to search for abbreviations.
+
+    Returns
+    -------
+    text: str
+        Text with compressed abbreviations.
+
+    Examples
+    ----------
+    >>> text = 'This is a test string (TS). I hope it is informative (inf).'
+    >>> compressed = compressall(text)
+    >>> print(compressed)
+    This is a TS (TS). I hope it is inf (inf).
```

Comment:
On a usage note, why not remove the abbreviations themselves; i.e., the sentence would become "This is a TS. I hope it is inf."?

Reply:
I'm not 100% sure about how to do it, but that is a great idea. We should do the same thing for expanding abbreviations as well. Do you want to open an issue so that it's on the to do list?